scripts/setup_mtk_optee.sh

```diff
@@ -0,0 +1,446 @@
+#!/bin/bash
+################################################################################
+# EDIT so these match your credentials/preferences                             #
+################################################################################
+DEV_PATH=$HOME/devel/mtk_optee
+
+# You only need to set these variables if you have access to the OPTEE_TEST
+# (requires a Linaro account and access to the git called optee_test.git)
+# If, in addition to the base OPTEE_TEST, you have access to the GlobalPlatform
+# "TEE Initial Configuration" test suite, you may add the tests by extracting
+# the test package in the current directory and run:
+#   $ export CFG_GP_PACKAGE_PATH=<path to the test suite directory>
+#   $ export CFG_GP_TESTSUITE_ENABLE=y
+#   $ ./setup_qemu_optee.sh
+#   $ ./build.sh
+#LINARO_USERNAME=firstname.lastname # Should _NOT_ contain @linaro.org.
+#HAVE_ACCESS_TO_OPTEE_TEST=1
+
+# If the downloaded toolchain can't execute it could be that you're in a
+# 64-bit system without required 32-bit libs
+# For Ubuntu 14.04 the following helps:
+# sudo apt-get install libc6:i386 libstdc++6:i386 libz1:i386
+
+################################################################################
+# Don't touch anything below this comment                                      #
+################################################################################
+set -e
+mkdir -p $DEV_PATH
+
+SRC_KERNEL=https://github.com/ibanezchen/linux-8173.git
```

Comment:
Is this a special kernel needed? I.e, is it impossible to use the upstream kernel?